### PR TITLE
[FEATURE] Improve dApps connection

### PR DESCRIPTION
**Pull request by andreahaku**
**Description**

Improves connection between dApps and MM mobile app

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Progresses #???

### CODE REVIEW

**Review by andrepimenta — APPROVED** [review]

File: app/core/SDKConnect.ts [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        this.timeout = BackgroundTimer.setTimeout(() => {
+          this.pause();
+        }, 20000);
+      }
+    }
+  },
+  async init() {
+    /*AsyncStorage.setItem('sdkConnections', JSON.stringify({}));
```
Comment (andrepimenta) [inline app/core/SDKConnect.ts:352]: Remove this

File: android/app/build.gradle [hunk 2]
```diff
@@ -160,8 +160,8 @@ android {
 		applicationId "io.metamask"
 		minSdkVersion rootProject.ext.minSdkVersion
 		targetSdkVersion rootProject.ext.targetSdkVersion
-		versionCode 939
-		versionName "5.5.0"
+		versionCode 951
+		versionName "5.5.99"
```
Comment (gantunesr) [inline android/app/build.gradle:164]: Revert before merging

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 3]
```diff
@@ -75,8 +105,54 @@ const ExperimentalSettings = ({ navigation, route }: Props) => {
...
+    </ActionModal>
+  );
+
   return (
     <ScrollView style={styles.wrapper}>
+      <View style={styles.setting}>
+        <View>
+          <Text style={styles.title}>{'MetaMask SDK Connections'}</Text>
+          <Text style={styles.desc}>
+            {'Clear all MetaMask SDK connections'}
```
Comment (gantunesr) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:145]: Add to strings file

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 4]
```diff
@@ -75,8 +105,54 @@ const ExperimentalSettings = ({ navigation, route }: Props) => {
...
+        </Text>
+      </View>
+    </ActionModal>
+  );
+
   return (
     <ScrollView style={styles.wrapper}>
+      <View style={styles.setting}>
+        <View>
+          <Text style={styles.title}>{'MetaMask SDK Connections'}</Text>
```
Comment (gantunesr) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:143]: Add to strings file

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 5]
```diff
@@ -75,8 +105,54 @@ const ExperimentalSettings = ({ navigation, route }: Props) => {
...
+          <Text style={styles.desc}>
+            {'Clear all MetaMask SDK connections'}
+          </Text>
+          <StyledButton
+            type="signingCancel"
+            onPress={toggleClearMMSDKConnectionModal}
+            containerStyle={styles.clearHistoryConfirm}
+          >
+            {'Clear MetaMask SDK connections'}
```
Comment (gantunesr) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:152]: Add to strings file

File: app/core/SDKConnect.ts [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+  approvedHosts[host] = hostname;
+
+  DefaultPreference.set('sdkApprovedHosts', JSON.stringify(approvedHosts));
+};
+
+const parseSource = (source: string) => {
+  if (source === 'web-desktop') return 'web-desktop';
+  if (source === 'web-mobile') return 'web-mobile';
+  if (source === 'nodejs') return 'nodejs';
+  if (source === 'unity') return 'unity';
```
Comment (gantunesr) [inline app/core/SDKConnect.ts:82]: A good improvement would be to add this values to a `Enum`

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -825,7 +825,7 @@
 				CODE_SIGN_ENTITLEMENTS = MetaMask/MetaMaskDebug.entitlements;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Manual;
-				CURRENT_PROJECT_VERSION = 939;
+				CURRENT_PROJECT_VERSION = 951;
```
Comment (gantunesr) [inline ios/MetaMask.xcodeproj/project.pbxproj:828]: Revert before merge

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -861,7 +861,7 @@
 					"\"$(SRCROOT)/MetaMask/System/Library/Frameworks\"",
 				);
 				LLVM_LTO = YES;
-				MARKETING_VERSION = 5.5.0;
+				MARKETING_VERSION = 5.5.99;
```
Comment (gantunesr) [inline ios/MetaMask.xcodeproj/project.pbxproj:864]: Revert before merge

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -892,7 +892,7 @@
 				CODE_SIGN_ENTITLEMENTS = MetaMask/MetaMask.entitlements;
 				CODE_SIGN_IDENTITY = "iPhone Distribution";
 				CODE_SIGN_STYLE = Manual;
-				CURRENT_PROJECT_VERSION = 939;
+				CURRENT_PROJECT_VERSION = 951;
```
Comment (gantunesr) [inline ios/MetaMask.xcodeproj/project.pbxproj:895]: Revert before merge

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -927,7 +927,7 @@
 					"\"$(SRCROOT)/MetaMask/System/Library/Frameworks\"",
 				);
 				LLVM_LTO = YES;
-				MARKETING_VERSION = 5.5.0;
+				MARKETING_VERSION = 5.5.99;
```
Comment (gantunesr) [inline ios/MetaMask.xcodeproj/project.pbxproj:930]: Revert before merge

File: locales/languages/en.json [hunk 11]
```diff
@@ -637,7 +637,9 @@
     "delete_data_status_description": "The current status is",
     "delete_metrics_error_title": "We are unable to delete this data right now.",
     "delete_metrics_error_description": "This request can't be completed right now due to an analytics system server issue, please try again later.",
-    "ok": "OK"
+    "ok": "OK",
+    "clear_sdk_connections_title": "Clear all MetaMask SDK Connections",
+    "clear_sdk_connections_text": "All connections will be cleared and dapps will need to request connection again"
```
Comment (gantunesr) [inline locales/languages/en.json:642]: dapps, Dapps, or dApps? 🤔

**Review by gantunesr — CHANGES_REQUESTED** [review]
Left some minor comments and there are tests failing

File: app/components/UI/ApproveTransactionReview/index.js [hunk 12]
```diff
@@ -438,6 +448,11 @@ class ApproveTransactionReview extends PureComponent {
         },
         unlimited_permission_requested: unlimited,
         referral_type: isDapp ? 'dapp' : transaction?.origin,
+        request_source: this.originIsMMSDKRemoteConn
+          ? 'MetaMask-SDK-Remote-Conn'
```
Comment (sethkfman) [inline app/components/UI/ApproveTransactionReview/index.js:452]: Should these be stored as constants?

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 13]
```diff
@@ -75,8 +106,56 @@ const ExperimentalSettings = ({ navigation, route }: Props) => {
     navigation.navigate('WalletConnectSessionsView');
   }, [navigation]);
 
+  const toggleClearMMSDKConnectionModal = () => {
+    setshowClearMMSDKConnectionsModal((show) => !show);
+  };
+
+  const clearMMSDKConnections = async () => {
+    SDKConnect.disconnectAll();
+    await AsyncStorage.setItem('sdkConnections', JSON.stringify({}));
```
Comment (sethkfman) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:115]: Should be converted to constants

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 14]
```diff
@@ -75,8 +106,56 @@ const ExperimentalSettings = ({ navigation, route }: Props) => {
...
+    await AsyncStorage.setItem('sdkApprovedHosts', JSON.stringify({}));
```
Comment (sethkfman) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:116]: Should be converted to constants

File: app/core/SDKConnect.ts [hunk 15]
```diff
@@ -0,0 +1,363 @@
...
+        }
+      }
+    } else if (appState === 'background' && !this.paused) {
+      if (Device.isIos()) {
+        BackgroundTimer.start();
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        this.timeout = setTimeout(() => {
+          this.pause();
+        }, 20000);
```
Comment (sethkfman) [inline app/core/SDKConnect.ts:343]: Consider making this a constant that describe the reason for the timeout

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request closed without merging**